packages/ripple-binary-codec/test/tx-encode-decode.test.js

```diff
@@ -48,7 +52,30 @@ describe('encoding and decoding tx_json', function () {
     })
     const encoded = encode(my_tx)
     const decoded = decode(encoded)
-    expect(my_tx).toEqual(decoded)
+    expect(my_tx).deep.equal(decoded)
+  })
+  test('can decode an transaction with an issued currency that evaluates to XRP', function () {
+    // Encoding is done prior, because this is disallowed during encoding with client libraries to avoid scam XRP tokens.
+    const expectedTx = {
+      TransactionType: 'TrustSet',
+      Flags: 0,
+      Sequence: 19,
+      LimitAmount: {
+        value: '200',
+        currency: 'XRP',
+        issuer: 'r9hEDb4xBGRfBCcX3E4FirDWQBAYtpxC8K',
+      },
+      Fee: '10',
+      SigningPubKey:
+        '023076CBB7A61837F1A23D4A3DD7CE810B694992EB0959AB9D6F4BB6FED6F8CC26',
+      TxnSignature:
+        '304502202D0CD77D8E765E3783C309CD663723B18406B7950A348A6F301492916A990FC70221008A76D586111205304F10ADEFDFDDAF804EF202D8CD1E492DC6E1AA8030EA1844',
+      Account: 'rPtfQWdcdhuL9eNeNv5YfmekSX3K7vJHbG',
+    }
+    const encoded =
+      '1200142200000000240000001363D5071AFD498D00000000000000000000000000005852500000000000585E1F3BD02A15D6185F8BB9B57CC60DEDDB37C168400000000000000A7321023076CBB7A61837F1A23D4A3DD7CE810B694992EB0959AB9D6F4BB6FED6F8CC267447304502202D0CD77D8E765E3783C309CD663723B18406B7950A348A6F301492916A990FC70221008A76D586111205304F10ADEFDFDDAF804EF202D8CD1E492DC6E1AA8030EA18448114FAFD540634705D7B1B9C8C31BFE2CC40326EB404'
+    const decoded = decode(encoded)
```

Comment:
Does rippled ban currencies that are equivalent to XRP? If that's true, then this should be fine. If not, then this should fail.

Reply:
Well we've got one on ledger which has this encoded value, so I think it doesn't (or at least historically didn't) which leads me to believe this should be fine.

Reply:
It wasn't XRP and shouldn't be XRP. It was `0x0000000000000000000000005852500000000000`

Reply:
Oh, you mean does rippled accept **encoded** `xrp` codes. Let me check.

Reply:
Submitting an issued currency with code `0x0000000000000000000000005852500000000000` results in temBAD_CURRENCY (While sending an issued currency with code `XRP` results in `Invalid native currency`) So they both error out, but give different messages. (Note that changing the 25 to 24 results in a successful issued currency trust set, so it's specifically if it decodes to XRP that creates the temBAD_CURRENCY as far as I can tell)

Reply:
So I think the answer is: Yes, rippled bans currencies that are equivalent to XRP

Reply:
Then yeah, we should still throw here.